astropy/units/quantity.py

```diff
@@ -1454,33 +1471,63 @@ def to_string(self, unit=None, precision=None, format=None, subfmt=None):
 
         if format not in formats:
             raise ValueError(f"Unknown format '{format}'")
-        elif format is None:
-            if precision is None:
-                # Use default formatting settings
-                return f"{self.value}{self._unitstr:s}"
-            else:
-                # np.array2string properly formats arrays as well as scalars
-                return (
-                    np.array2string(self.value, precision=precision, floatmode="fixed")
-                    + self._unitstr
+
+        format_spec = formatter if isinstance(formatter, str) else None
+
+        if format is None:
+            if format_spec is not None:
+
+                def formatter(value):
+                    return builtins.format(value, format_spec)
+
+            if callable(formatter):
+                formatter = {"all": formatter}
+
+            return (
+                np.array2string(
+                    self.value,
+                    precision=precision,
+                    floatmode="fixed",
+                    formatter=formatter,
                 )
+                + self._unitstr
+            )
 
         # else, for the moment we assume format="latex" or "latex_inline".
 
         # Set the precision if set, otherwise use numpy default
         pops = np.get_printoptions()
-        format_spec = f".{precision if precision is not None else pops['precision']}g"
+        if format_spec is None:
+            format_spec = (
+                f".{precision if precision is not None else pops['precision']}g"
+            )
 
-        def float_formatter(value):
-            return Latex.format_exponential_notation(value, format_spec=format_spec)
+        # Use default formatters
+        if formatter is None or isinstance(formatter, str):
+            if self.dtype.kind == "c":  # Complex default latex formatter
+                # Disallowed sign operations for complex numbers
```

Comment:
Hmm, this would also change the format for the real part, which is not the idea - you just need to change it for complex only. It is actually somewhat tricky to get completely right. For instance, python does
```
In [2]: format(1+1j, "<+20.5f")
Out[2]: '+1.00000+1.00000j   '
```
So, the width argument should apply to the whole result, not each one individually.

To be honest, I'm happy to ignore that for now for `latex`, where width has less meaning anyway (all those spaces get ignored!). But we should apply the changed format to imaginary only.